### ISSUE TITLE
Secure backend's endpoints with correct permission checks

### ISSUE THREAD

**Issue by petertrr** [analysis-dev/save-cloud#417]
Should be done after #470 and #495
- Only some endpoints are accessible for unauthorized users (is configurable from `WebSecurityConfig` like in #498)
- Only project's owner can delete stuff and change description and submit executions

List of required roles:
- `ROLE_ADMIN` - can delete anything, can view anything
- `ROLE_PROJECT_OWNER` - this is the user, that created the project and is linked to it in the DB
- `ROLE_PROJECT_ADMIN` - can be added by project owner, can start and manage executions, can't delete execution and project

Implementation:
- Create roles in DB
- Create roles hierarchy for spring security
- Protect methods according to roles: https://www.baeldung.com/spring-security-method-security#5-method-security-meta-annotation, https://www.baeldung.com/spring-security-create-new-custom-security-expression

**Comment by petertrr** [comment]
Thoughts from #424:
[* Convert backend into a resource server to properly consume tokens? https://docs.spring.io/spring-security/site/docs/5.2.1.RELEASE/reference/htmlsingle/#webflux-oauth2-resource-server. Then, hardcoded endpoints can be removed from gateway's security config, and backend can check if endpoint is available for a current user. After this PR is merged, need to be implemented in a separate issue. Also can use CircuitBreaker filter to redirect to login page on status 403
